File: CS-Examples/14_Hyperlinks/AddHyperlinkToText.cs
using System;
using System.Windows.Forms;
using Spire.Xls;

namespace AddHyperlinkToText
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            // Create a workbook
			Workbook workbook = new Workbook();

            // Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\CommonTemplate1.xlsx");

            // Get the first sheet
            Worksheet sheet = workbook.Worksheets[0];

            // Add url link
            HyperLink UrlLink = sheet.HyperLinks.Add(sheet.Range["D10"]);
            // Set display text
            UrlLink.TextToDisplay = sheet.Range["D10"].Text;
            // Set url link type
            UrlLink.Type = HyperLinkType.Url;
            // Set url address
            UrlLink.Address = "http://en.wikipedia.org/wiki/Chicago";

            //Add email link
            HyperLink MailLink = sheet.HyperLinks.Add(sheet.Range["E10"]);
            // Set display text
            MailLink.TextToDisplay = sheet.Range["E10"].Text;
            // Set mail link type
            MailLink.Type = HyperLinkType.Url;
            // Set mail address
            MailLink.Address = "mailto:Amor.Aqua@gmail.com";

            // Specify the file name for the resulting Excel file
            string output = "AddHyperlinkToText.xlsx";

            // Save the workbook to the specified file in Excel 2010 format
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the Excel file
            ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: CS-Examples/14_Hyperlinks/AddHyperlinkToText.vb
Imports Spire.Xls

Namespace AddHyperlinkToText

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load an existing Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\CommonTemplate1.xlsx")

            ' Get the first worksheet from the loaded workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Add a hyperlink to a cell (D10) in the worksheet
            Dim UrlLink As HyperLink = sheet.HyperLinks.Add(sheet.Range("D10"))
            UrlLink.TextToDisplay = sheet.Range("D10").Text
            UrlLink.Type = HyperLinkType.Url
            UrlLink.Address = "http://en.wikipedia.org/wiki/Chicago"

            ' Add another hyperlink to a cell (E10) in the worksheet
            Dim MailLink As HyperLink = sheet.HyperLinks.Add(sheet.Range("E10"))
            MailLink.TextToDisplay = sheet.Range("E10").Text
            MailLink.Type = HyperLinkType.Url
            MailLink.Address = "mailto:Amor.Aqua@gmail.com"

            ' Specify the filename for the resulting Excel file
            Dim output As String = "AddHyperlinkToText.xlsx"

            ' Save the Workbook object to the specified file path in Excel 2013 format
            workbook.SaveToFile(output, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the Excel file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
